istioctl/cmd/wait.go

```diff
@@ -241,28 +243,15 @@ func getAndWatchResource(ictx context.Context) *watcher {
 		version := targetSchema.Resource().Version()
 		resource := collectionParts[3]
 		r := dclient.Resource(schema.GroupVersionResource{Group: group, Version: version, Resource: resource}).Namespace(namespace)
-		obj, err := r.Get(context.TODO(), nameflag, metav1.GetOptions{})
+		watch, err := r.Watch(context.TODO(), metav1.ListOptions{})
```

Comment:
Add `"fieldSelector": "metadata.name=" + NAME,` and you can avoid watching all resources when we only want one of them